packages/protocol/test/governance/voting/lockedgold.ts

```diff
@@ -771,4 +798,753 @@ contract('LockedGold', (accounts: string[]) => {
+          assertEqualBN(
+            await lockedGold.getAccountTotalGovernanceVotingPower(delegatee1),
+            delegatedAmount - amountToRevoke
+          )
+          await assertDelegatorDelegateeAmounts(
+            delegator,
+            delegatee1,
+            percentsToDelegate - percentageToRevoke,
+            delegatedAmount - amountToRevoke
+          )
+
+          assertEqualBN(
+            await lockedGold.totalDelegatedCelo(delegatee1),
+            delegatedAmount - amountToRevoke
+          )
+        })
+
+        it('should emit the CeloDelegatedRevoked event', async () => {
+          assert.equal(resp.logs.length, 1)
+          const log = resp.logs[0]
+          assertLogMatches2(log, {
+            event: 'CeloDelegatedRevoked',
+            args: {
+              delegator: delegator,
+              delegatee: delegatee1,
+              percent: percentageToRevoke,
+              amount: amountToRevoke,
+            },
+          })
+        })
+
+        it('should revoke votes correctly when delegatee not voting (delegated to 2 accounts)', async () => {
+          await lockedGold.delegateGovernanceVotes(delegatee2, percentsToDelegate)
+
+          assertEqualBN(
+            await lockedGold.getAccountTotalGovernanceVotingPower(delegatee1),
+            delegatedAmount - amountToRevoke
+          )
+          await assertDelegatorDelegateeAmounts(
+            delegator,
+            delegatee1,
+            percentsToDelegate - percentageToRevoke,
+            delegatedAmount - amountToRevoke
+          )
+
+          assertEqualBN(
+            await lockedGold.getAccountTotalGovernanceVotingPower(delegatee2),
+            delegatedAmount
+          )
+          await assertDelegatorDelegateeAmounts(
+            delegator,
+            delegatee2,
+            percentsToDelegate,
+            delegatedAmount
+          )
+
+          assertEqualBN(
+            await lockedGold.totalDelegatedCelo(delegatee1),
+            delegatedAmount - amountToRevoke
+          )
+          assertEqualBN(await lockedGold.totalDelegatedCelo(delegatee2), delegatedAmount)
+        })
+
+        describe('When delegator1 locked more gold + delegator2 also delegated to delegatee1', () => {
+          beforeEach(async () => {
+            assertEqualBN(
+              await lockedGold.getAccountTotalGovernanceVotingPower(delegatee1),
+              delegatedAmount - amountToRevoke
+            )
+            await lockedGold.delegateGovernanceVotes(delegatee1, percentsToDelegate, {
+              from: delegator2,
+            })
+            assertEqualBN(
+              await lockedGold.getAccountTotalGovernanceVotingPower(delegatee1),
+              delegatedAmount + (delegatedAmount - amountToRevoke)
+            )
+            await lockedGold.lock({ value: value, from: delegator })
+          })
+
+          describe('When revoking percentage such as that with newly locked amount it would decrease below zero', () => {
```

Comment:
It is not percentage that drops below zero, but rather amount.

Scenario is as follows:

1. Delegator will delegate 10% (100) of his total amount (1000)
2. Delegator will lock another 1000
3. Delegator will revoke 10% 

The delegated amount is not updated during lock and until someone will call updateDelegatedAmount the delegated amount will still remain 100. When revoke with 8% would be called then 8% is counted from the original new amount -> 160. Now if we would try to remove 160 we would get exception even though this is absolutely valid scenario. In this case we need to make sure not to go below zero.

This scenario is no longer up to date since I added updateDelegatedAmount call into revokeDelegatedGovernanceVotes (thanks for this suggestion ! )